packages/content-management/favorites/favorites_public/src/favorites_client.ts

```diff
@@ -40,21 +42,45 @@ export class FavoritesClient<Metadata extends object | void = void>
   constructor(
     private readonly appName: string,
     private readonly favoriteObjectType: string,
-    private readonly deps: { http: HttpStart; usageCollection?: UsageCollectionStart }
+    private readonly deps: {
+      http: HttpStart;
+      security: SecurityServiceStart;
+      usageCollection?: UsageCollectionStart;
+    }
   ) {}
 
+  private isAvailableCached?: boolean;
+  public async isAvailable(): Promise<boolean> {
+    if (typeof this.isAvailableCached === 'boolean') return this.isAvailableCached;
+
+    try {
+      const user = await this.deps.security.authc.getCurrentUser();
```

Comment:
@azasypkin, could you please take a look if this looks good? I want to disable the favorites feature usage on the client if `user.profile_uid` is not available

Reply:
It looks good to me. As we discussed over Slack, there’s a slight chance that the cached values might not reflect the actual state in certain cases (e.g., re-authentication through other tabs that don’t involve a logout). However, for this specific scenario, I think the risk is negligible: if the actual value for a new session differs from the cached one, users will either hit an error when saving something as a favorite or notice that the functionality is disabled/not present. In either case, a page refresh will help them.

By the way, the Core also exposes [getUserProfile$ through api_provider and eui_provider](https://github.com/search?q=repo%3Aelastic%2Fkibana%20getUserProfile%24&type=code), which might give you the same result, though I’m not sure if you can leverage it from this package.

Reply:
userProfile.getUserProfile$ might be a better fit, I didn't realize it exist in core. will give it a try, thanks

Reply:
took a quick look, it sees that there is a theoretical race condition that getUserProfile at first emits null while profile might still be loading. 

https://github.com/elastic/kibana/blob/714ba675f78098c20271906a12f59da27c45cbf5/x-pack/plugins/security/public/account_management/user_profile/user_profile_api_client.ts#L34


Would be cool to have this enabled exposed instead, but it is not available in core 

https://github.com/elastic/kibana/blob/714ba675f78098c20271906a12f59da27c45cbf5/x-pack/plugins/security/public/account_management/user_profile/user_profile_api_client.ts#L46-L50